src/link/Elf.zig

```diff
@@ -1581,40 +1581,74 @@ fn parseLdScript(self: *Elf, lib: SystemLib) ParseError!void {
 }
 
 pub fn validateEFlags(self: *Elf, file_index: File.Index, e_flags: elf.Word) !void {
-    if (self.first_eflags == null) {
-        self.first_eflags = e_flags;
-        return; // there isn't anything to conflict with yet
-    }
-    const self_eflags: *elf.Word = &self.first_eflags.?;
-
+    const target = self.getTarget();
     switch (self.getTarget().cpu.arch) {
         .riscv64 => {
-            if (e_flags != self_eflags.*) {
-                const riscv_eflags: riscv.RiscvEflags = @bitCast(e_flags);
-                const self_riscv_eflags: *riscv.RiscvEflags = @ptrCast(self_eflags);
-
-                self_riscv_eflags.rvc = self_riscv_eflags.rvc or riscv_eflags.rvc;
-                self_riscv_eflags.tso = self_riscv_eflags.tso or riscv_eflags.tso;
-
-                var any_errors: bool = false;
-                if (self_riscv_eflags.fabi != riscv_eflags.fabi) {
-                    any_errors = true;
-                    try self.addFileError(
-                        file_index,
-                        "cannot link object files with different float-point ABIs",
-                        .{},
-                    );
-                }
-                if (self_riscv_eflags.rve != riscv_eflags.rve) {
-                    any_errors = true;
-                    try self.addFileError(
-                        file_index,
-                        "cannot link object files with different RVEs",
-                        .{},
-                    );
-                }
-                if (any_errors) return error.LinkFailure;
+            const features = target.cpu.features;
+            const flags: riscv.Eflags = @bitCast(e_flags);
+            var any_errors: bool = false;
+
+            // For an input object to target an ABI that the target CPU doesn't have enabled
+            // is invalid, and will throw an error.
+
+            // Invalid when
+            // 1. The input uses C and we do not.
+            if (flags.rvc and !std.Target.riscv.featureSetHas(features, .c)) {
+                any_errors = true;
+                try self.addFileError(
+                    file_index,
+                    "cannot link object file targeting the C ABI without have the C feature enabled",
```

Comment:
```suggestion
                    "cannot link object file targeting the C ISA extension without have the 'c' feature enabled",
```

I think "C ABI" is going to be misleading to the average user here, so probably better to just use this terminology.

Reply:
Hmm, I could agree with this. Although I feel "ISA extension" is too complicatedand could just be "feature".

Reply:
Minor typo: should also be "without hav**ing** the 'c' feature enabled"

Reply:
thanks, ive pushed both. changed from "ABI" to "feature" since I think that makes the most sense if we're not using "ABI".